Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp

```diff
@@ -238,6 +238,33 @@ TEST_F(TerrainPhysicsColliderComponentTest, TerrainPhysicsColliderGetHeightsRetu
     AZ::Vector2 mockHeightResolution = AZ::Vector2(1.0f);
     NiceMock<UnitTest::MockTerrainDataRequests> terrainListener;
     ON_CALL(terrainListener, GetTerrainHeightQueryResolution).WillByDefault(Return(mockHeightResolution));
+    ON_CALL(terrainListener, ProcessHeightsFromRegion).WillByDefault(
```

Comment:
If the MockTerrainDataRequests is defined, then all calls to functions in the TerrainDataRequestBus will be routed through the Mock implementation. The implementations being tested are in the perPositionCallback() defined in TerrainPhysicsColliderComponent. For this reason, we need to provide a mock implementation for the ProcessRegion functions here and in other cases below.